### PR TITLE
first draft of generating code with file templates

### PR DESCRIPTION
**Pull request by ztatlock**
@Ptival @d1jang  still adding some features, but here's the core of the template based generator.

### CODE REVIEW

**Comment by ztatlock** [comment]
@Ptival @d1jang can you check this code out now.

I only have one more feature to add before I'm willing to merge: adding Ptival's comments regexp thing for the genCoq version.

**Comment by ztatlock** [comment]
@Ptival @d1jang OK I think this is ready to merge. Can you check it out and let me know what you think?

**Comment by d1jang** [comment]
It looks quite good! Do I need to merge this change with my ifexp branch?

**Comment by ztatlock** [comment]
Not yet.

**Comment by Ptival** [comment]
Sorry, went to bed at 9 because of major toothache and medicine kicking in :crying_cat_face: 

Will take a look at it this morning.

PS: Type a colon and get access to stupid emoticons! :camel: :fr:

**Comment by ztatlock** [comment]
Bwahaha, like the Caml ;)

Hope your dents are feeling better!

File: compiler/src/genCoq.ml [hunk 1]
```diff
@@ -803,78 +384,90 @@ Proof.
...
+  comp kstate_vars hands kstate_vars fall_through
+
+let coq_of_kernel_subs s =
+  let m = gen_tag_map s in
+  let (xch_chan, exchanges) = s.exchange in
+  let kstate_vars =
+    s.var_decls |> List.map fst |> String.concat " "
+  in
+  List.map (fun (f, r) ->
+    (Str.regexp ("(\\* *__" ^ f ^ "__ *\\*)"), r))
```
Comment (Ptival) [inline compiler/src/genCoq.ml:396]: Oulala ! :smile_cat:

**Comment by Ptival** [comment]
Nothing seems blatantly wrong. I trust you for the Makefile changes. :) Seems good to me.

**Pull request merged**